src/Evolution/Systems/Burgers/BoundaryConditions/DirichletAnalytic.cpp

```diff
@@ -20,7 +38,72 @@ DirichletAnalytic::get_clone() const {
   return std::make_unique<DirichletAnalytic>(*this);
 }
 
-void DirichletAnalytic::pup(PUP::er& p) { BoundaryCondition::pup(p); }
+std::optional<std::string> DirichletAnalytic::dg_ghost(
+    const gsl::not_null<Scalar<DataVector>*> u,
+    const gsl::not_null<tnsr::I<DataVector, 1, Frame::Inertial>*> flux_u,
+    const std::optional<
+        tnsr::I<DataVector, 1, Frame::Inertial>>& /*face_mesh_velocity*/,
+    const tnsr::i<DataVector, 1, Frame::Inertial>& /*normal_covector*/,
+    const tnsr::I<DataVector, 1, Frame::Inertial>& coords,
+    [[maybe_unused]] const double time) const {
+  call_with_dynamic_type<void, tmpl::append<Burgers::Solutions::all_solutions,
+                                            Burgers::AnalyticData::all_data>>(
+      analytic_prescription_.get(),
+      [&coords, &time, &u](const auto* const analytic_solution_or_data) {
+        if constexpr (is_analytic_solution_v<
+                          std::decay_t<decltype(*analytic_solution_or_data)>>) {
+          *u = get<Burgers::Tags::U>(analytic_solution_or_data->variables(
+              coords, time, tmpl::list<Burgers::Tags::U>{}));
+        } else {
+          *u = get<Burgers::Tags::U>(analytic_solution_or_data->variables(
+              coords, tmpl::list<Burgers::Tags::U>{}));
+          (void)time;
+        }
+      });
+  flux_impl(flux_u, *u);
+  return {};
+}
+
+void DirichletAnalytic::fd_ghost(
```

Comment:
Note: These functions are just moved from the hpp and switched to using the member variable analytic prescription